### ISSUE TITLE
vectorAdd_profiled fails to compile with VS2019

### ISSUE THREAD

**Issue by Q-Minh** [eyalroz/cuda-api-wrappers#345]
The example target `vectorAdd_profiled` has the error `cuda-api-wrappers/src\cuda/nvtx/profiling.hpp(339): error : no instance of function template "cuda::profiling::detail_::name_host_thread" matches the argument list`.

**Comment by eyalroz** [comment]
I have no idea why the compiler says that... does it provide any more information?

**Comment by eyalroz** [comment]
Try it now.

**Comment by Q-Minh** [comment]
So, I made some modifications to the code to make the example compile on my end, but it looks like I can't submit a PR on this repo, so I will just list my changes here. I basically replaced every instance of `#if __cplusplus >= 201703L` with `#if __cplusplus >= 201703L || _MSVC_LANG >= 201703L` in the code, specifically in the files:
- `examples/modified_cuda_samples/memMapIPCDrv/child.cpp`
- `examples/other/jitify.cpp`
- `examples/other/jitify/type_name.hpp`
- `cuda/api/link.hpp`
- `cuda/nvrtc/detail/marshalled_options.hpp`
- `cuda/api/module.hpp`
- `cuda/nvrtc/types.hpp`
because some files use `std::string_view`, `std::to_char`, or `std::filesystem`.

Then, because `std::size_t` is defined to be `unsigned long long` on some platforms, for example my platform, you can add corresponding overloads `append_to_current(unsigned long long)` in the `cuda/nvrtc/detail/marshalled_options.hpp` file, otherwise compilation will fail, since the call to `append_to_current(cuda::size_t{})` is ambiguous, and could match many different overloads.

**Comment by eyalroz** [comment]
@Q-Minh : To submit a PR, you clone the repo, create a branch, then submit the PR.

**Comment by Q-Minh** [comment]
> @Q-Minh : To submit a PR, you clone the repo, create a branch, then submit the PR.

That's what I did, but I get an HTTP 403 when I attempt to push the branch onto the remote.

**Comment by Q-Minh** [comment]
Let me try it again, it might have to do with https vs ssh, since I have multiple github accounts on the same system.

**Comment by Q-Minh** [comment]
@eyalroz Yeah, I can't push the branch onto the remote, unfortunately. Might be some problem on my end. Do you mind making the changes for me? The fist change is a pretty quick string search and replace, and the second one is just writing up 2 overloads. :)

**Comment by eyalroz** [comment]
Does it work?

**Comment by Q-Minh** [comment]
Yes, it does.
